crates/epaint/src/image.rs

```diff
@@ -101,6 +101,65 @@ impl ColorImage {
         Self { size, pixels }
     }
 
+    pub fn from_rgba_premultiplied(size: [usize; 2], rgba: &[u8]) -> Self {
+        assert_eq!(size[0] * size[1] * 4, rgba.len());
+        let pixels = rgba
+            .chunks_exact(4)
+            .map(|p| Color32::from_rgba_premultiplied(p[0], p[1], p[2], p[3]))
+            .collect();
+        Self { size, pixels }
+    }
+
+    /// A view of the underlying data as `&[u8]`
+    pub fn as_raw(&self) -> &[u8] {
+        bytemuck::cast_slice(&self.pixels)
+    }
+
+    /// A view of the underlying data as `&mut [u8]`
+    pub fn as_raw_mut(&mut self) -> &mut [u8] {
+        bytemuck::cast_slice_mut(&mut self.pixels)
+    }
+
+    /// Reinterpret the underlying data as `Vec<u8>` instead of `Vec<egui::Color32>`. Useful for interoperating with other crates such as [`image`](crates.io/crates/image). This
+    /// is a no-copy, but uses unsafe code internally. For a safer alternative, you can use [`Self::as_raw`] and manually copy into a new collection.
+    pub fn into_raw(mut self) -> Vec<u8> {
+        let ratio = std::mem::size_of::<Color32>() / std::mem::size_of::<u8>();
+        let length = self.pixels.len() * ratio;
+        let capacity = self.pixels.capacity() * ratio;
+        let ptr = self.pixels.as_mut_ptr() as *mut u8;
+        std::mem::forget(self.pixels);
+        unsafe { Vec::from_raw_parts(ptr, length, capacity) }
```

Comment:
I believe this is unsound. From reading the `bytemuck` docs, you can only do this if the alignment of the source and target type is the same, which it isn't

I suggest removing this method. Users can to it themselves if they wish to.

Reply:
Done. I assume the cast_slice's are still fine, as Color32 is annotated with bytemuck::Pod?